test/unit/history/invoke.test.js

```diff
@@ -0,0 +1,23 @@
+import History from '../../../src/history'
+import Microcosm from '../../../src/microcosm'
+
+describe('History::invoke', function() {
+
+  it('it does invoke a repo that no longer exists', function() {
+    let parent = new Microcosm()
+    let child = parent.fork()
+
+    parent.on('change', () => child.teardown())
+
+    jest.spyOn(child, 'release')
+
+    parent.addDomain('test', {
+      getInitialState() {
+        return false
+      }
+    })
+
+    expect(child.release).not.toHaveBeenCalled()
+  })
```

Comment:
As for `release`, `release` is an internal Microcosm method for sending out a change event. It fires a change, then it invokes Effects with the related action.

This allows a tree of microcosms (parent + forks) to all resolve an action, then handle change subscriptions when every microcosm is up to date.